providers/ethereum-provider/src/EthereumProvider.ts

```diff
@@ -538,10 +543,14 @@ export class EthereumProvider implements IEthereumProvider {
   protected async loadPersistedSession() {
     if (!this.session) return;
     const chainId = await this.signer.client.core.storage.getItem(`${this.STORAGE_KEY}/chainId`);
-    this.setChainIds(
-      chainId ? [this.formatChainId(chainId)] : this.session.namespaces[this.namespace].accounts,
-    );
-    this.setAccounts(this.session.namespaces[this.namespace].accounts);
+
+    // cater to both inline & nested namespace formats
+    const namespace = this.session.namespaces[this.namespace]
```

Comment:
Should this be reversed? i.e try `this.session.namespaces[`${this.namespace}:${chainId}`]` first and then do general namespace?